EU4toV2/Source/V2World/V2World.cpp

```diff
@@ -88,10 +91,62 @@ historicalData(sourceWorld.getHistoricalData())
 	LOG(LogLevel::Info) << "*** Goodbye, Vicky 2, and godspeed. ***";
 }
 
-void V2::World::addAcceptedCultures()
+void V2::World::dropStates()
+{
+	// We have dropped a lot of cores recently. It's time to drop states to territory status if they do not contain any own cores.
+	for (const auto& country: countries)
+	{
+		for (const auto& state: country.second->getStates())
+		{
+			auto hasCore = false;
+			for (const auto& province: state->getProvinces())
+			{
+				if (province->hasCore(country.first)) hasCore = true;
+			}
+			if (!hasCore) state->setProvincesAsTerritories();
+		}
+	}
+}
+
+void V2::World::dropCores()
+{
+	// This is quicker if we first build a country/culture cache and then check against it then iterate through 
+	// every province and do multiple unneeded checks.
+
+	std::map<std::string, std::set<std::string>> theCache; // tag, accepted cultures cache
+	std::set<std::string> deadCache; // dead countries where we need to preserve cores as they have no cultures.
+	
+	for (const auto& country: countries)
+	{
+		theCache[country.first] = country.second->getAcceptedCultures();
+		theCache[country.first].insert(country.second->getPrimaryCulture());
+		if (country.second->getProvinces().empty()) deadCache.insert(country.first);
+	}
+	
+	for (auto& province: provinces)
+	{
+		const auto dominantCulture = province.second->getDominantCulture();
+		std::set<std::string> survivingCores;
+		for (const auto& core: province.second->getCores())
+		{
+			// Dead countries take priority.
+			if (deadCache.count(core)) continue; // not touching this.
```

Comment:
The comment for the deadCache says its to preserve cores, but this will ensure the dead cores are not in the surviving cores list and dropped. Which way is it?

Reply:
It's the special brainFart() function.